xrmocap/model/mvp/projattn.py

```diff
@@ -0,0 +1,281 @@
+# yapf: disable
+
+import logging
+import math
+import torch
+import torch.nn.functional as F
+import warnings
+from torch import nn
+from torch.autograd import Function
+from torch.autograd.function import once_differentiable
+from torch.nn.init import constant_, xavier_uniform_
+from typing import Union
+from xrprimer.utils.log_utils import get_logger
+
+try:
+    import Deformable as DF  # noqa F401
+    has_deformable = True
+    import_exception = ''
+except (ImportError, ModuleNotFoundError):
+    has_deformable = False
+    import traceback
+    import_exception = traceback.format_exc()
+
+# yapf: enable
+
+
+class ProjAttn(nn.Module):
+
+    def __init__(self,
+                 logger: Union[None, str, logging.Logger],
+                 d_model=256,
+                 n_levels=4,
+                 n_heads=8,
+                 n_points=4,
+                 projattn_pos_embed_mode='use_rayconv'):
+        """Projective Attention Module Modified from DeformableConvV2 and
+        Deformable DETR https://github.com/chengdazhi/Deformable-
+        Convolution-V2-PyTorch/tree/pytorch_1.0.0
+        https://github.com/fundamentalvision/Deformable-DETR.
+
+        Args:
+            d_model (int, optional):
+                Hidden dimension. Defaults to 256.
+            n_levels (int, optional):
+                Number of feature levels. Defaults to 4.
+            n_heads (int, optional):
+                Number of attention heads. Defaults to 8.
+            n_points (int, optional):
+                Number of sampling points per attention head
+                per feature level. Defaults to 4.
+            projattn_pos_embed_mode (str, optional):
+                The positional embedding mode of projective
+                attention. Defaults to 'use_rayconv'.
```

Comment:
Write out all possible values? e.g., The mode includes 'use_rayconv' and  'use_2d_coordconv'. The former denotes xxx, while the latter means xxx. Defaults to  'use_rayconv'.

Reply:
Added in [d5025b7](https://github.com/openxrlab/xrmocap/pull/21/commits/d5025b700c504f02e48cdaae421d19a4d3d2c8d4).